PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStorageWithUploadPathPanel.kt

```diff
@@ -0,0 +1,127 @@
+
+    private val jobUploadStorageTitle = "Job Upload Storage"
+    private val uploadPathLabel = JLabel("Upload Path")
+    private val uploadPathField = JTextField().apply {
+        isEditable = false
+        border = BorderFactory.createEmptyBorder()
+    }
+    val storagePanel = SparkSubmissionJobUploadStoragePanel()
+    private val hideableJobUploadStoragePanel = HideableTitledPanel(jobUploadStorageTitle, true, storagePanel, false)
+
+    private val layoutPlan = listOf(
+            Place(uploadPathLabel, buildConstraints(0).apply { row = 0 }), Place(uploadPathField, buildConstraints(1).apply { row = 0 }),
+            Place(hideableJobUploadStoragePanel, baseConstraints().apply {
+                row = 1
+                colSpan = 2
+                hSizePolicy = GridConstraints.SIZEPOLICY_WANT_GROW
+                fill = GridConstraints.FILL_HORIZONTAL
+
+            })
+    )
+
+    init {
+        layout = GridLayoutManager(layoutPlan.last().gridConstraints.row + 1, colTemplate.size)
+        layoutPlan.forEach { (component, gridConstrains) -> add(component, gridConstrains) }
+    }
+
+    override fun getData(data: SparkSubmitJobUploadStorageModel) {
+        // Component -> Data
+        data.selectedStorageTypeIndex = storagePanel.storageTypeComboBox.selectedIndex
+        data.errorMsg = storagePanel.errorMessage
+        data.uploadPath = uploadPathField.text
+        when (storagePanel.storageTypeComboBox.selectedItem) {
+            StorageType.AzureBlob.title -> {
+                data.storageAccountType = SparkSubmitStorageType.Blob.toString()
+                data.storageAccount = storagePanel.azureBlobCard.storageAccountField.text.trim()
+                data.storageKey = storagePanel.azureBlobCard.storageKeyField.text.trim()
+                data.containersModel = storagePanel.azureBlobCard.storageContainerComboBox.model as DefaultComboBoxModel
+                data.selectedContainer = storagePanel.azureBlobCard.storageContainerComboBox.selectedItem as? String
+            }
+            StorageType.ClusterDefaultStorage.title -> {
+                data.storageAccountType = SparkSubmitStorageType.DefaultStorageAccount.toString()
+            }
+            StorageType.SparkInteractiveSession.title -> {
+                data.storageAccountType = SparkSubmitStorageType.SparkInteractiveSession.toString()
+            }
+        }
+    }
+
+    override fun setData(data: SparkSubmitJobUploadStorageModel) {
+        // data -> Component
+        val applyData: () -> Unit = {
+            storagePanel.storageTypeComboBox.selectedIndex = data.selectedStorageTypeIndex
+            storagePanel.errorMessage = data.errorMsg
+            storagePanel.storageAccountType = data.storageAccountType
+            uploadPathField.text = data.uploadPath
+            if (data.selectedStorageTypeIndex != -1) {
```

Comment:
I found this:

> -1 if no item is selected or if the currently selected item is not in the list

Should we to check `getModel().getSelectedItem() == null`?

Reply:
I think we should get selected item from `data`(`getItemAt(data.selectedStorageTypeIndex)`) rather than from UI (`getModel().getSelectedItem()`) since  selected item in UI might not be the validation item.

---
In reply to: [223566228](https://github.com/Microsoft/azure-tools-for-java/pull/2137#discussion_r223566228) [](ancestors = 223566228)